runtime/interface.cpp

```diff
@@ -601,6 +607,20 @@ void f$fastcgi_finish_request(int64_t exit_code) {
       break;
     }
     case QUERY_TYPE_HTTP: {
+      // Run custom headers handler before body processing
+      if (headers_custom_handler_function && !headers_sent) {
+        headers_sent = true;
+        headers_custom_handler_function();
+        // In `headers_custom_handler_function` might be happened `f$flush` invocation
+        // First of all, we have to accumulate response data from buffers
+        ob_total_buffer = ob_merge_buffers();
+        // If a flushing is happened, send response without headers
+        if (php_worker.has_value() && php_worker->flushed_http_connection) {
+          string const raw_response = oub[ob_total_buffer].str();
+          http_set_result(nullptr, 0, raw_response.c_str(), raw_response.size(), static_cast<int32_t>(exit_code));
+          php_assert (0);
+        }
```

Comment:
Also I can't understand why this logic was changed.
As far as I can see, If flush was not invoked from header callback, we just go by usual path without changes.
If flush was invoked, it means header callback was invoked too, and headers already was sent, and then we go by usual old path too. Do I miss something?

Reply:
Callback invocation moved into more appropriate place, fixed